workflow/controller/operator.go

```diff
@@ -492,29 +492,20 @@ func (woc *wfOperationCtx) persistUpdates() {
 			return
 		}
 		woc.log.Info("Re-applying updates on latest version and retrying update")
-		wf, err := woc.reapplyUpdate(wfClient)
+		wf, err = woc.reapplyUpdate(wfClient)
 		if err != nil {
 			woc.log.Infof("Failed to re-apply update: %+v", err)
 			return
 		}
-		woc.wf = wf
-	} else {
-		woc.wf = wf
 	}
 
+	woc.controller.latch.Update(wf.UID, wf.ResourceVersion)
```

Comment:
After object is updated,  we can get the updated resource version, call update method of latch with it.